starboard/shared/starboard/player/filter/testing/audio_decoder_test.cc

```diff
@@ -833,6 +874,95 @@ TEST_P(AudioDecoderTest, PartialAudio) {
   }
 }
 
+TEST_P(AudioDecoderTest, ResetInEndOfStreamStateWithEndOfStream) {
+  ASSERT_NO_FATAL_FAILURE(WriteSingleInput(0));
+  WriteEndOfStream();
+  ASSERT_NO_FATAL_FAILURE(DrainOutputs());
+
+  ResetDecoder();
+  WriteEndOfStream();
+  ASSERT_NO_FATAL_FAILURE(DrainOutputs());
+  ASSERT_TRUE(decoded_audios_.empty());
+  ASSERT_NO_FATAL_FAILURE(AssertOutputFormatValid());
+}
+
+TEST_P(AudioDecoderTest, ResetInEndOfStreamStateWithSingleInput) {
```

Comment:
nit: I think the name "ResetAfterEndOfStreamWithSingleInput" should be ok, same for next test case.